tests/slurm_plugin/slurm_resources/test_slurm_resources.py

```diff
@@ -287,6 +287,65 @@ def test_slurm_node_is_up(node, expected_output):
     assert_that(node.is_up()).is_equal_to(expected_output)
 
 
+@pytest.mark.parametrize(
+    "node, expected_output",
+    [
+        (StaticNode("queue1-st-c5xlarge-1", "nodeip", "nodehostname", "IDLE+CLOUD+POWER", "queue1"), False),
+        (
+            StaticNode("queue1-st-c5xlarge-1", "nodeip", "nodehostname", "IDLE+CLOUD+MAINTENANCE+RESERVED", "queue1"),
+            True,
+        ),
+        (
+            StaticNode(
+                "queue1-st-c5xlarge-1",
+                "nodeip",
+                "nodehostname",
+                "DOWN+CLOUD+MAINTENANCE+POWERED_DOWN+RESERVED",
+                "queue1",
+            ),
+            True,
+        ),
+        (
+            DynamicNode("queue1-dy-c5xlarge-1", "nodeip", "nodehostname", "IDLE+CLOUD+RESERVED", "queue1"),
+            True,
+        ),
+        (DynamicNode("queue1-dy-c5xlarge-1", "nodeip", "nodehostname", "RESERVED", "queue1"), True),
```

Comment:
is it possible to have a state with MAINTENANCE but no RESERVED?

Reply:
Nope, this is not possible, "maint" is a flag passed at reservation creation. It's not possible to set a node in maintenance without being in reserved state.